### PR TITLE
Clean up code

### PR DESCRIPTION
**Pull request by JasonGrace2282**
Most of these were automatically fixed by running pre-commit (#1659 ). 

Note that most of these should be pretty safe, as `ruff` was run _without_ the `--unsafe-fixes` flag, and `ruff`has been designed around not modifying the resulting runtime code so most of it should run the same

99% of changes fall into one of the 3 categories;
* Typos
* `"...".format(...)` to `f"..."`
* End of line endings to unix end of line character

EDIT: I removed changes from all vendor files to reduce the diff (it went from a 3,000 character diff to a 500 character diff). If the maintainers of Ion want vendor files to be fixed, I can make another PR for that.

### CODE REVIEW

File: intranet/apps/eighth/serializers.py [hunk 1]
```diff
@@ -223,9 +222,9 @@ def fetch_activity_list_with_metadata(self, block):
 
         sponsors_dict = EighthSponsor.objects.values_list("id", "user_id", "first_name", "last_name", "show_full_name")
 
-        all_sponsors = dict(
+        all_sponsors = {
```
Comment (JasonGrace2282) [inline intranet/apps/eighth/serializers.py:225]: This change was manually made and is incorrect

**Review by JasonGrace2282 — COMMENTED** [review]
From looking through, I caught one error (and that was made by me xD)

File: intranet/apps/eighth/serializers.py [hunk 2]
```diff
@@ -223,9 +222,9 @@ def fetch_activity_list_with_metadata(self, block):
...
             (sponsor[0], {"user_id": sponsor[1], "name": sponsor[2] + " " + sponsor[3] if sponsor[4] else sponsor[3]}) for sponsor in sponsors_dict
```
Comment (JasonGrace2282) [inline intranet/apps/eighth/serializers.py:226]:
```suggestion
             sponsor[0]: {"user_id": sponsor[1], "name": sponsor[2] + " " + sponsor[3] if sponsor[4] else sponsor[3]}) for sponsor in sponsors_dict
```

**Comment by JasonGrace2282** [comment]
Turns out if you modify a chunk of the codebase, you also have to "fix" the linting errors in that area. I fixed (or in somee  cases, put a bandaid: these should (ideally) be fixed in a follow-up PR) [these](https://github.com/tjcsl/ion/actions/runs/9102038320/job/25020706045) pylint warnings.

File: intranet/apps/eighth/tests/admin/test_admin_groups.py [hunk 3]
```diff
@@ -531,23 +531,23 @@ def test_delete_empty_groups(self):
         self.make_admin()
 
         # Create users
-        userA = get_user_model().objects.get_or_create(username="userA")[0]
+        users = get_user_model().objects.get_or_create(username="users")[0]
         userB = get_user_model().objects.get_or_create(username="userB")[0]
```
Comment (krishnans2006) [inline intranet/apps/eighth/tests/admin/test_admin_groups.py:535]: Oops

**Review by krishnans2006 — CHANGES_REQUESTED** [review]

File: intranet/apps/eighth/tests/admin/test_admin_groups.py (hunk 3 above)
Comment (JasonGrace2282) [reply intranet/apps/eighth/tests/admin/test_admin_groups.py]: Reverted this change, it shouldn't be here :)

File: intranet/apps/eighth/management/commands/remove_withdrawn_students.py [hunk 4]
```diff
@@ -41,11 +41,11 @@ def handle(self, *args, **options):
                 "eighth/emails/withdrawn_students.txt",
                 "eighth/emails/withdrawn_students.html",
                 data,
-                "Withdrawn Students on {}".format(str(timezone.localdate())),
```
Comment (NotFish232) [inline intranet/apps/eighth/management/commands/remove_withdrawn_students.py:44]: !s is unnecessary

**Review by NotFish232 — CHANGES_REQUESTED** [review]

File: intranet/apps/schedule/models.py [hunk 5]
```diff
@@ -145,7 +145,7 @@ def end_time(self):
         return self.day_type.end_time
 
     def __str__(self):
-        return "{}: {}".format(str(self.date), self.day_type)
+        return f"{self.date!s}: {self.day_type}"
```
Comment (NotFish232) [inline intranet/apps/schedule/models.py:148]: same !s issue

File: intranet/apps/search/views.py [hunk 6]
```diff
@@ -201,9 +201,9 @@ def get_search_results(q, admin=False):
 
 def do_activities_search(q):
     filter_query = get_query(q, ["name", "description"])
-    entires = EighthActivity.objects.filter(filter_query).order_by("name")
+    entries = EighthActivity.objects.filter(filter_query).order_by("name")
     final_entires = []
```
Comment (NotFish232) [inline intranet/apps/search/views.py:205]: should also be final_entries

**Review by NotFish232 — CHANGES_REQUESTED** [review]

File: intranet/apps/search/views.py (hunk 6 above)
Comment (NotFish232) [reply intranet/apps/search/views.py:205]: same with the other changes later in the file

**Review by NotFish232 — APPROVED** [review]
LGTM But let Alan look through it too first

File: intranet/apps/groups/models.py [hunk 7]
```diff
@@ -65,4 +65,4 @@ class GroupProperties(models.Model):
     student_visible = models.BooleanField(default=False)
 
     def __str__(self):
-        return "{}".format(self.group)
+        return f"{self.group}"
```
Comment (alanzhu0) [inline intranet/apps/groups/models.py:68]: :/

File: intranet/apps/lostfound/models.py [hunk 8]
```diff
@@ -11,7 +11,7 @@ class LostItem(models.Model):
     found = models.BooleanField(default=False)
 
     def __str__(self):
-        return "{}".format(self.title)
+        return f"{self.title}"
```
Comment (alanzhu0) [inline intranet/apps/lostfound/models.py:14]: :/

File: intranet/apps/lostfound/models.py [hunk 9]
```diff
@@ -26,7 +26,7 @@ class FoundItem(models.Model):
     retrieved = models.BooleanField(default=False)
 
     def __str__(self):
-        return "{}".format(self.title)
+        return f"{self.title}"
```
Comment (alanzhu0) [inline intranet/apps/lostfound/models.py:29]: :/

File: intranet/apps/notifications/models.py [hunk 10]
```diff
@@ -18,7 +18,7 @@ def gcm_token_sha256(self):
         return hashlib.sha256(self.gcm_token.encode()).hexdigest()
 
     def __str__(self):
-        return "{}".format(self.user)
+        return f"{self.user}"
```
Comment (alanzhu0) [inline intranet/apps/notifications/models.py:21]: :/

File: intranet/apps/eighth/management/commands/absence_email.py [hunk 11]
```diff
@@ -19,7 +19,7 @@ def handle(self, *args, **options):
 
         for signup in absences:
             if log:
-                self.stdout.write("{}".format(signup))
+                self.stdout.write(f"{signup}")
```
Comment (alanzhu0) [inline intranet/apps/eighth/management/commands/absence_email.py:22]: :/

File: intranet/apps/eighth/management/commands/signup_status_email.py [hunk 12]
```diff
@@ -45,27 +45,27 @@ def handle(self, *args, **options):
...
                 if log:
-                    self.stdout.write("Block {} on {} is not today ({}).".format(next_blocks[0], blk_date, today))
+                    self.stdout.write(f"Block {next_blocks[0]} on {blk_date} is not today ({today}).")
                 return
 
         if log:
-            self.stdout.write("{}".format(next_blocks))
-            self.stdout.write("{}".format(options))
-            self.stdout.write("{}".format(users))
+            self.stdout.write(f"{next_blocks}")
```
Comment (alanzhu0) [inline intranet/apps/eighth/management/commands/signup_status_email.py:59]: :/

File: intranet/apps/eighth/views/admin/activities.py [hunk 13]
```diff
@@ -190,7 +188,7 @@ def edit_activity_view(request, activity_id):
     for g in activity.groups_allowed.all():
         group = {}
         group["id"] = g.id
-        group["name"] = "{}".format(g)
+        group["name"] = f"{g}"
```
Comment (alanzhu0) [inline intranet/apps/eighth/views/admin/activities.py:191]: :/

File: intranet/apps/eighth/views/admin/attendance.py [hunk 14]
```diff
@@ -178,8 +178,8 @@ def no_signups_roster(request, block_id):
 
         for user in unsigned:
             row = []
-            row.append("{}".format(block.id))
-            row.append("{}".format(block))
+            row.append(f"{block.id}")
```
Comment (alanzhu0) [inline intranet/apps/eighth/views/admin/attendance.py:181]: :/

File: intranet/apps/eighth/views/admin/blocks.py [hunk 15]
```diff
@@ -340,7 +340,7 @@ def print_block_rosters_view(request, block_id):
         try:
             block = EighthBlock.objects.get(id=block_id)
             schacts = EighthScheduledActivity.objects.filter(block=block)
-            schacts = sorted(schacts, key=lambda x: "{}".format(x.get_true_sponsors()))
+            schacts = sorted(schacts, key=lambda x: f"{x.get_true_sponsors()}")
```
Comment (alanzhu0) [inline intranet/apps/eighth/views/admin/blocks.py:343]: :/

File: intranet/apps/files/views.py [hunk 16]
```diff
@@ -174,13 +174,13 @@ def files_type(request, fstype=None):
             try:
                 sftp.chdir(host_dir)
             except exceptions as e:
-                if "NoSuchFile" in "{}".format(e):
+                if "NoSuchFile" in f"{e}":
```
Comment (alanzhu0) [inline intranet/apps/files/views.py:177]: :/

File: intranet/apps/files/views.py [hunk 17]
```diff
@@ -435,8 +435,8 @@ def rmtree(sftp, path):
         except PermissionError:
             messages.error(request, "You are not allowed to delete this " + ("folder" if is_directory else "file") + "!")
         except exceptions as e:
-            messages.error(request, "{}".format(e))
-        return redirect("/files/{}?dir={}".format(fstype, os.path.dirname(filepath)))
+            messages.error(request, f"{e}")
```
Comment (alanzhu0) [inline intranet/apps/files/views.py:438]: :/

File: intranet/apps/schedule/notifications.py [hunk 18]
```diff
@@ -16,9 +16,9 @@ def period_start_end_data(request):
     blocks = ctx["sched_ctx"]["blocks"]
     point, block = at_period_point(blocks)
     if point == 1:
-        return {"title": "{} has started".format(block.name), "text": "{}".format(block)}
+        return {"title": f"{block.name} has started", "text": f"{block}"}
     elif point == 2:
-        return {"title": "{} has ended".format(block.name), "text": "{}".format(block)}
+        return {"title": f"{block.name} has ended", "text": f"{block}"}
```
Comment (alanzhu0) [inline intranet/apps/schedule/notifications.py:21]: :/

File: intranet/apps/schedule/views.py [hunk 19]
```diff
@@ -399,9 +399,9 @@ def admin_daytype_view(request, daytype_id=None):
 
             if "delete" in request.POST:
                 daytype = get_object_or_404(DayType, id=daytype_id)
-                name = "{}".format(daytype)
+                name = f"{daytype}"
```
Comment (alanzhu0) [inline intranet/apps/schedule/views.py:402]: :/

File: intranet/apps/search/views.py [hunk 20]
```diff
@@ -114,7 +114,7 @@ def query(q, admin=False):
 
             # fix grade, because LDAP only stores graduation year
             if cat == "grade" and is_entirely_digit(val):
-                val = "{}".format(Grade.year_from_grade(int(val)))
+                val = f"{Grade.year_from_grade(int(val))}"
```
Comment (alanzhu0) [inline intranet/apps/search/views.py:117]: :/

File: intranet/apps/seniors/models.py [hunk 21]
```diff
@@ -129,7 +129,7 @@ class Senior(models.Model):
     major_sure = models.BooleanField(default=False)
 
     def __str__(self):
-        return "{}".format(self.user)
+        return f"{self.user}"
```
Comment (alanzhu0) [inline intranet/apps/seniors/models.py:132]: :/

File: intranet/apps/users/management/commands/import_groups.py [hunk 22]
```diff
@@ -20,17 +20,17 @@ def handle(self, *args, **options):
                     mappings[gid] = gexist[0]
                 else:
                     ngrp = Group.objects.create(name=gname)
-                    self.stdout.write("Created group {} with new id {}, old id {}".format(gname, ngrp.id, gid))
+                    self.stdout.write(f"Created group {gname} with new id {ngrp.id}, old id {gid}")
                     mappings[gid] = ngrp
 
-        self.stdout.write("{}".format(mappings))
-        with open("groups_static.csv", "r", encoding="utf-8") as staticopen:
+        self.stdout.write(f"{mappings}")
```
Comment (alanzhu0) [inline intranet/apps/users/management/commands/import_groups.py:26]: :/

**Review by alanzhu0 — CHANGES_REQUESTED** [review]
Not your fault but

**Review by alanzhu0 — APPROVED** [review]

**Pull request merged**